build-configuration/bom/pom.xml

```diff
@@ -179,11 +179,23 @@
                 <artifactId>infinispan-commons-test</artifactId>
                 <version>${version.infinispan}</version>
             </dependency>
-            <dependency>
-                <groupId>org.infinispan</groupId>
-                <artifactId>infinispan-core</artifactId>
-                <version>${version.infinispan}</version>
-            </dependency>
+           <dependency>
+              <groupId>org.infinispan</groupId>
+              <artifactId>infinispan-component-annotations</artifactId>
+              <version>${version.infinispan}</version>
+           </dependency>
+           <!-- TODO should be here? -->
```

Comment:
Is this something to worry about? Guessing not since it is labeled as optional.

Reply:
I just read a bit on optional vs provided and apparently declaring a dependency as optional in the `<dependencyManagement>` section doesn't actually make it optional :)

I'm going to leave it here without the optional and the TODO, and make it optional in the core pom instead.